### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by Zxun2**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -166,44 +166,51 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
-### Command Autocompletion, Real-Time Input Validation and Recommendation
+### Command Recommendation and Autocompletion
 
-Autocompletion and Recommendation are features that can greatly enhance the user experience when using the application.
-To support this feature, we have implemented a `CommandRecommendationEngine` that aims to accurately predict the set of
-words that the user intends to type based off a preset of words. These words are the in-built commands and argument
-prefixes.
+Autocompletion and command recommendation are crucial features that help to improve the user experience when interacting 
+with an application. By predicting the set of words that the user intends to type based on a preset of words such as 
+in-built commands and argument prefixes, the `CommandRecommendationEngine` helps to save users time and effort while
```
Comment (Zxun2) [inline docs/DeveloperGuide.md:173]:
```suggestion
in-built commands and field prefixes, the `CommandRecommendationEngine` helps to save users time and effort while 
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -166,44 +166,51 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
-### Command Autocompletion, Real-Time Input Validation and Recommendation
+### Command Recommendation and Autocompletion
 
-Autocompletion and Recommendation are features that can greatly enhance the user experience when using the application.
-To support this feature, we have implemented a `CommandRecommendationEngine` that aims to accurately predict the set of
-words that the user intends to type based off a preset of words. These words are the in-built commands and argument
-prefixes.
+Autocompletion and command recommendation are crucial features that help to improve the user experience when interacting 
+with an application. By predicting the set of words that the user intends to type based on a preset of words such as 
+in-built commands and field prefixes, the `CommandRecommendationEngine` helps to save users time and effort while
```
Comment (wz2k) [inline docs/DeveloperGuide.md:173]:
```suggestion
with our application. By predicting the set of words that the user intends to type based on a preset of words such as 
in-built commands and field prefixes, the `CommandRecommendationEngine` helps to save user's time and effort while 
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -166,44 +166,51 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
-2. `autocompleteCommand` -- autocompletes the user input by replacing user input with the recommended command.
-3. `registerCommandInfo` -- registers the command, turning on command recommendation.
+To provide autocompletion, the `CommandRecommendationEngine` uses an event handler attached to the `commandTextField` 
+which listens for the `TAB` event. When triggered, it autocompletes the user's input by replacing it with the recommended values.
 
-#### Autocompletion
+On the other hand, to provide command recommendations, the engine uses the longest prefix matching algorithm. This 
+algorithm first **identifies** the related command based on the user's input and then **verifies** the fields 
+associated with it. For instance, when the user types "add_e" in the command prompt, the engine will first recommend
+"add_elderly", follow by the related fields. If there are ambiguity in the recommendations, the decision is to return
```
Comment (wz2k) [inline docs/DeveloperGuide.md:184]:
```suggestion
"add_elderly", followed by the related fields. If there are ambiguities in the recommendations, the decision is to return 
```

**Review by wz2k — APPROVED** [review]
LGTM. Need change some sentences.

**Pull request merged**